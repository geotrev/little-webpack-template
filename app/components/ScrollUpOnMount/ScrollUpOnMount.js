import React from "react"

/*
 * NOTE: This is a hack to mimic page navigation.
 *       Simply import as a component like you would anywhere,
 *       and the page will auto-scroll to the top like a "new" page
 */

export default class ScrollUpOnMount extends React.Component {
  componentDidMount() {
    window.scrollTo(0, 0)
  }

  render() {
    return null
  }
}
