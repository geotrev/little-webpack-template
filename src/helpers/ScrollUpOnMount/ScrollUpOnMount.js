import React from "react"

/*
 * NOTE: This is a hack to mimic page navigation.
 * Simply import as a component like you would anywhere.
 */

export default class ScrollUpOnMount extends React.Component {
  componentDidMount() {
    window.scrollTo(0, 0)
  }

  render() {
    return null
  }
}
