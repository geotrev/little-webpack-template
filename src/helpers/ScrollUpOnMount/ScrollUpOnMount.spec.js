import React from "react"
import ScrollUpOnMount from "./ScrollUpOnMount"

global.scrollTo = jest.fn()

describe("<ScrollUpOnMount />", () => {
  it("matches snapshot", () => {
    const wrapper = mount(<ScrollUpOnMount />)
    expect(wrapper).toMatchSnapshot()
  })

  it("calls scrollTo with correct values", () => {
    const wrapper = mount(<ScrollUpOnMount />)
    expect(window.scrollTo).toBeCalledWith(0, 0)
  })
})
