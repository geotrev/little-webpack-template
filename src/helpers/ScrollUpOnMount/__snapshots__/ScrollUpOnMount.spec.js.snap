// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScrollUpOnMount /> matches snapshot 1`] = `<ScrollUpOnMount />`;
